libgalois/include/katana/MemoryPolicy.h

```diff
@@ -53,31 +53,53 @@ class KATANA_EXPORT MemoryPolicy {
   double kill_used_ratio_threshold() const {
     return thresholds_.kill_used_ratio_threshold;
   }
+  count_t kill_self_oom_threshold() const {
+    return thresholds_.kill_self_oom_threshold;
+  }
+  count_t high_pressure_oom_threshold() const {
+    return thresholds_.high_pressure_oom_threshold;
+  }
 
 private:
   count_t physical_;
   Thresholds thresholds_;
 };
 
-/// Memory policy that just tries to avoid the OOM killer and does little to evict
-/// unused data from memory
+/// Memory policy that just tries to avoid the OOM killer.  Unfortunately, it is aggressive about
+/// dumping memory when the OOM score is high, which can be an over reaction.
 class KATANA_EXPORT MemoryPolicyMinimal : public MemoryPolicy {
 public:
   MemoryPolicyMinimal();
-  count_t ReclaimForMemoryPressure(
-      count_t active, count_t standby) const override;
-  bool MemoryPressureHigh(count_t active, count_t standby) const override;
-  bool KillSelfForLackOfMemory(count_t active, count_t standby) const override;
+  count_t ReclaimForMemoryPressure(count_t standby) const override;
+  bool IsMemoryPressureHigh(count_t standby) const override;
+  bool KillSelfForLackOfMemory(count_t standby) const override;
 };
 
 /// Memory policy that prioritizes performance, i.e., it uses memory aggressively
 class KATANA_EXPORT MemoryPolicyPerformance : public MemoryPolicy {
 public:
   MemoryPolicyPerformance();
-  count_t ReclaimForMemoryPressure(
-      count_t active, count_t standby) const override;
-  bool MemoryPressureHigh(count_t active, count_t standby) const override;
-  bool KillSelfForLackOfMemory(count_t active, count_t standby) const override;
+  count_t ReclaimForMemoryPressure(count_t standby) const override;
+  bool IsMemoryPressureHigh(count_t standby) const override;
+  bool KillSelfForLackOfMemory(count_t standby) const override;
+};
+
+/// Minimize use of memory, but take free memory when it is available.
+class KATANA_EXPORT MemoryPolicyMeek : public MemoryPolicy {
+public:
+  MemoryPolicyMeek();
+  count_t ReclaimForMemoryPressure(count_t standby) const override;
+  bool IsMemoryPressureHigh(count_t standby) const override;
+  bool KillSelfForLackOfMemory(count_t standby) const override;
+};
+
+/// Do nothing to ever shed memory.  This will OOM if we occupy too much memory.
+class KATANA_EXPORT MemoryPolicyNull : public MemoryPolicy {
+public:
+  MemoryPolicyNull();
+  count_t ReclaimForMemoryPressure(count_t standby) const override;
+  bool IsMemoryPressureHigh(count_t standby) const override;
+  bool KillSelfForLackOfMemory(count_t standby) const override;
```

Comment:
I need to tell the compiler that I am overriding them because they are pure virtual in the base class.  But if they weren't pure virtual and I omitted their type signature in the derived class, I'd get the base implementation, which I don't want.

Reply:
Right, I was just wondering if you could override and define the functions in .cpp without having to redefine the interface here, if that makes sense. But perhaps in order for the vtable lookup to work, it needs it defined in the headers as well? Just seems repetitive.

Reply:
I'm not a c++ expert, but I think it is necessary.